Common/src/Pollster/Pollster.cs

```diff
@@ -199,12 +202,27 @@ await taskHandler.ReleaseTask(CancellationToken.None)
                 return;
               }
 
-              await taskHandler.ExecuteTask(combinedCts.Token)
+              var workerConnectionCts = new CancellationTokenSource();
+              var postConnectionCts = new CancellationTokenSource();
+              var errorConnectionCts = new CancellationTokenSource();
+              combinedCts.Token.Register(() =>
```

Comment:
You should register for the CancellationToken of main loop instead. This one includes a cancellationToken from the message (is it still relevant anyway with AMQP ?).